clang/lib/Parse/ParseCXXInlineMethods.cpp

```diff
@@ -978,6 +978,20 @@ bool Parser::ConsumeAndStoreFunctionPrologue(CachedTokens &Toks) {
       } else {
         break;
       }
+      // Pack indexing
+      if (getLangOpts().CPlusPlus26 && Tok.is(tok::ellipsis) &&
```

Comment:
FWIW it would be really sweet if we could get this in older language modes to implement `std::tuple`. I don't know whether the current version is ABI compatible, but IMO it would definitely be worth looking into it.

Reply:
I'd suggested offline that we bring @ldionne in and see if he can make a case for doing so.  However, this IS somewhat of a 'breaking change' for a minor case, so we have to balance that.

Reply:
@philnik777 This would require GCC to also support that as an extension in C++11 (and above). Otherwise we wouldn't be able to use it in our tuple implementation. If we had such an extension, I suspect compilation times for any tuple-heavy code would go down quite a bit. In fact the most interesting part is that we might be able to simplify how we store elements inside the tuple -- right now we have to use some kind of base class trick because that's how we manage to perform indexing afterwards, but that's expensive.

Reply:
Would a change like that be possible without being an ABI break?  I might bring @AaronBallman into this here discussion and see what we think here.  I see value in enabling it here and hoping GCC does someday, but not sure I'm in favor of it, unless this is something you could use ANYWAY without their support.

Another thing is perhaps we could ask @jicama what GCC intends to do here (or at least for his thoughts/see if we can influence them)?  @jwakely might have something to say as well.

Reply:
I think it might be possible to maintain the same ABI we have right now while changing how it's implemented, yes. It probably requires a bit of thought but it's probably possible. But there's certainly no point in any of this if GCC doesn't add this extension, from my point of view at least.

Reply:
> @philnik777 This would require GCC to also support that as an extension in C++11 (and above). Otherwise we wouldn't be able to use it in our tuple implementation.

Yeah, obviously. But if one implementation adds it, it becomes a lot more likely that others do it too. I also suspect that libstdc++ may also really like this.



> Would a change like that be possible without being an ABI break?

I don't know, but IMO it would even be worth just adding it in ABIv2 if we can get it as an extension. I expect this to bring a _lot_ of improvement in terms of compile times and debug binary size.